src/Host.cpp

```diff
@@ -502,6 +503,34 @@ void Host::loadPackageInfo()
     }
 }
 
+void Host::createModuleBackup(const QString &filename, const QString &saveName)
+{
+    QString time = QDateTime::currentDateTime().toString("yyyy-MM-dd#HH-mm-ss");
+    QFile::copy(filename, saveName + time);
+}
+
+void Host::writeModules(const QStringList &entry, const QString &moduleName, QString &filename)
+{
+    if (filename.endsWith(QStringLiteral("mpackage"), Qt::CaseInsensitive) || filename.endsWith(QStringLiteral("zip"), Qt::CaseInsensitive)) {
+        filename = mudlet::getMudletPath(mudlet::profilePackagePathFileName, mHostName, moduleName);
+    }
+    auto writer = new XMLexport(this);
+    writers.insert(filename, writer);
+    writer->writeModuleXML(moduleName, filename);
+    updateModuleZips(entry[0], moduleName);
+}
+
+void Host::waitForHostXmlSave()
+{
+    auto myWriters = writers;
```

Comment:
This is needed so we wait for the host xml to save before we close the Host/Client.
It was in the original code as well I just took it out from waitForProfileSave and made it, its own function so I can use it somewhere else as well.

Reply:
I mean the `auto myWriters` bit?

Reply:
Oh that bit. 
I had crashes happening (sometimes) if I didn't do it this way.

The reason was that because of the async nature of this, in some cases writers (and futures) exactly got deleted right during the for loop was happening.

I tried some other ways to prevent that (like if writer/future == nullptr ...)
but the only way which prevented crashes was to put writers/futures into a variable right before the for loop was happening.

Not sure if in some cases this already caused crashes in Mudlet 4.13 (or earlier)

Reply:
OK makes sense, could you add a comment explaining it? Since off the shelf the code looks a bit odd